src/main/java/com/configcat/FetchResponse.java

```diff
@@ -22,24 +25,42 @@ public boolean isFailed() {
         return this.status == Status.FAILED;
     }
 
-    public Config config() {
-        return this.config;
+    public boolean isFetchTimeUpdatable() {
+        return fetchTimeUpdatable;
+    }
+
+    public Entry entry() {
+        return this.entry;
+    }
+
+    public String error() {
+        return this.error;
     }
 
-    FetchResponse(Status status, Config config) {
+    FetchResponse(Status status, Entry entry, String error, boolean fetchTimeUpdatable) {
         this.status = status;
-        this.config = config;
+        this.entry = entry;
+        this.error = error;
+        this.fetchTimeUpdatable = fetchTimeUpdatable;
     }
 
-    public static FetchResponse fetched(Config config) {
-        return new FetchResponse(Status.FETCHED, config == null ? Config.empty : config);
+    public static FetchResponse fetched(Entry entry) {
+        return new FetchResponse(Status.FETCHED, entry == null ? Entry.EMPTY : entry, null, false);
     }
 
     public static FetchResponse notModified() {
-        return new FetchResponse(Status.NOT_MODIFIED, Config.empty);
+        return new FetchResponse(Status.NOT_MODIFIED, Entry.EMPTY, null, false);
+    }
+
+    public static FetchResponse notModified(boolean fetchTimeUpdatable) {
```

Comment:
I think the `fetchTimeUpdatable` argument is unnecessary as it's always `true` in case of `notModified`.

Reply:
Yep, right now only called with `true` and the other constructor not used, but I think it's better if in all case (modified, fail) the default is `false`  and has to be called with an argument to override it.
Also, I'm thinking about the name to change it to `forceFetchTimeUpdate`, because that's happens in these cases. What do you think?

Reply:
I mean only the parameter of `notModified(boolean fetchTimeUpdatable)` is unnecessary because it'll be always called with true. The backing field and the param of `failed(boolean fetchTimeUpdatable)` are ok.